indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java

```diff
@@ -116,4 +129,72 @@ public int compare(WindowedDataSegment s1, WindowedDataSegment s2)
   {
     return new DatasourceRecordReader();
   }
+
+  private final JobConf dummy = new JobConf();
+  private final FileInputFormat fio = new FileInputFormat()
+  {
+    @Override
+    protected boolean isSplitable(FileSystem fs, Path filename)
+    {
+      return false;
+    }
+
+    @Override
+    public org.apache.hadoop.mapred.RecordReader getRecordReader(
+        org.apache.hadoop.mapred.InputSplit inputSplit, JobConf jobConf, Reporter reporter
+    ) throws IOException
+    {
+      throw new UnsupportedOperationException();
+    }
+  };
+
+  private DatasourceInputSplit toDataSourceSplit(List<WindowedDataSegment> segments)
+  {
+    String[] locations = null;
+    try {
+      locations = getFrequentLocations(segments);
+    }
+    catch (Exception e) {
+      // ignore
+    }
+    return new DatasourceInputSplit(segments, locations);
+  }
+
+  private String[] getFrequentLocations(List<WindowedDataSegment> segments) throws IOException
```

Comment:
would be nice to have unit tests for this inside DatasourceInputFormatTest for this.